numpy/lib/arraypad.py

```diff
@@ -1406,10 +1406,15 @@ def pad(array, pad_width, mode, **kwargs):
             newmat = _append_min(newmat, pad_after, chunk_after, axis)
 
     elif mode == 'reflect':
-        if narray.size == 0:
-            raise ValueError("There aren't any elements to reflect in `array`")
-
         for axis, (pad_before, pad_after) in enumerate(pad_width):
+            if newmat.shape[axis] == 0:
```

Comment:
Why newmat and not narray as was used previously?

Reply:
In fact, I don't see how the error here can be hit - newmat is the size after padding. Are you sure you're catching the right ValueError?

Reply:
newmat is a copy of narray and I do the check before padding. The error will be hit if the axis it empty and the padding is non-zero. I might be wrong of course, so if you would insist on using `narray`, I do not mind.

Reply:
Since newmat.shape changes each iteration, I'd prefer narray. You're right though, both would work

Reply:
Changed, thanks